packages/launchpad/src/checklist/index.tsx

```diff
@@ -1,41 +1,42 @@
 import classNames from 'classnames';
-import ChecklistItem from '../checklist-item';
-import { Task } from '../types';
-
+import { Children, cloneElement, type FC, type ReactElement } from 'react';
+import { Placeholder as ChecklistPlaceholder } from '../checklist-item';
+import type { Props as CheckListItemProps } from '../checklist-item';
 import './style.scss';
 
 interface ChecklistProps {
-	tasks: Task[] | null;
 	makeLastTaskPrimaryAction?: boolean;
+	children?: ReactElement< CheckListItemProps > | ReactElement< CheckListItemProps >[];
 }
 
-const Checklist = ( { tasks, makeLastTaskPrimaryAction }: ChecklistProps ) => {
+const Checklist: FC< ChecklistProps > = ( {
+	children,
+	makeLastTaskPrimaryAction,
+}: ChecklistProps ) => {
 	return (
 		<ul
 			className={ classNames( 'checklist__tasks', {
 				'checklist__has-primary-action': makeLastTaskPrimaryAction,
 			} ) }
 			aria-label="Launchpad Checklist"
 		>
-			{ tasks &&
-				tasks.map( ( task: Task, index: number ) => (
-					<ChecklistItem
-						key={ task.id }
-						task={ task }
-						isPrimaryAction={ makeLastTaskPrimaryAction && index === tasks.length - 1 }
-					/>
-				) ) }
+			{ Children.map( children || [], ( child, index ) => {
+				if ( index === Children.count( children ) - 1 ) {
```

Comment:
fixed 4d0f969

Reply:
Side note:
I agree with the complexity, but I think the problem here is to have a flag that should be used on the parent component and the children. 

Probably is possible to solve it with CSS,  I just changed it because I found a bug when we set the flag on the parent but we don't set on the children.